### PR TITLE
Split out service entities

### PR DESCRIPTION
**Pull request by balloob**
## Proposed change

While ensuring that notify entities are not shown on the generated dashboard, I realized that they are also grouped under "control" on device info page. This PR splits out notify and Assist-related domains to their own cards on the device info page.

CleanShot 2024-06-16 at 13 32 57@2x

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
	- Added categorization for new service-related entities, including "conversation," "notify," "stt," and "tts."
	- Updated device configuration page to include "service" category.
  
- **Improvements**
	- Enhanced hidden domain list to dynamically include service-related entities.
	- Added new translation for "service" in the English language file.

### CODE REVIEW

File: src/panels/config/devices/ha-config-device-page.ts [hunk 1]
```diff
@@ -190,18 +190,31 @@ export class HaConfigDevicePage extends LitElement {
...
-      const result = groupBy(entities, (entry) =>
-        entry.entity_category
-          ? entry.entity_category
-          : computeDomain(entry.entity_id) === "event"
-            ? "event"
-            : SENSOR_ENTITIES.includes(computeDomain(entry.entity_id))
-              ? "sensor"
-              : "control"
-      ) as Record<
+      const result = groupBy(entities, (entry) => {
```
Comment (balloob) [inline src/panels/config/devices/ha-config-device-page.ts:193]: rewrote this to avoid having to compute `domain` three times

**Comment by frenck** [comment]
I'm not sure if "Services" is the thing we should communicate in terms of UX. While this might hold true for a notification service, it becomes pretty confusing considering we have more things called services in our UI/ecosystem. For example, a switch/light has services too, why wouldn't these be marked as such?

Can we think of another grouping we could do? For example: "Notifications" or "Communications" or something like that?

**Comment by balloob** [comment]
APIs ?

File: src/panels/lovelace/common/generate-lovelace-config.ts [hunk 2]
```diff
@@ -35,18 +35,16 @@ import { ButtonsHeaderFooterConfig } from "../header-footer/types";
 const HIDE_DOMAIN = new Set([
   "automation",
   "configurator",
-  "conversation",
   "device_tracker",
   "event",
   "geo_location",
   "notify",
```
Comment (balloob) [inline src/panels/lovelace/common/generate-lovelace-config.ts:41]:
```suggestion
```

**Comment by frenck** [comment]
> APIs ?

One of the approaches we quickly discussed during our frontend standup, was to maybe split it in two groups.

1. Notification or Notifiers (notify entities)
2. Assist (tts, sst, conversation)

As notify really doesn't match well with the rest.

File: src/common/const.ts [hunk 3]
```diff
@@ -98,7 +99,7 @@ export const FIXED_DOMAIN_ICONS = {
   calendar: mdiCalendar,
   climate: mdiThermostat,
   configurator: mdiCog,
-  conversation: mdiMicrophoneMessage,
+  conversation: mdiForumOutline,
```
Comment (balloob) [inline src/common/const.ts:102]: this one was duplicate with STT.

**Comment by balloob** [comment]
Updated per comments.

**Review by silamon — APPROVED** [review]

**Pull request merged**